ci-operator/step-registry/ibmcloud/deprovision/vpc/ibmcloud-deprovision-vpc-commands.sh

```diff
@@ -101,6 +101,48 @@ echo "Using region: ${region}  resource_group: ${resource_group} vpc: ${vpc_name
 echo "DEBUG" "Removing the vpc ${vpc_name} ..."
 delete_vpc "${vpc_name}"
 
+hasError=0
+echo "DEBUG" "Checking the remains resources and the last_operation of the resources in ${resource_group}..."
+count=$(eval "${IBMCLOUD_CLI}" resource service-instances --type all -g ${resource_group} --output JSON | jq '.[]|.name' | wc -l)
+if [ "${count}" -gt 0 ]; then
+  echo "ERROR: has remain resource ..."
+  "${IBMCLOUD_CLI}" resource service-instances --type all -g ${resource_group} --output JSON | jq '.[]|.name+" "+.type+" "+.resource_id+" "+.crn'
+  "${IBMCLOUD_CLI}" resource service-instances --type all -g ${resource_group} --output JSON | jq '.[]|.last_operation'
+  hasError=1
+fi
+
+CLUSTER_FILTER="${NAMESPACE}-${UNIQUE_HASH}"
+echo "DEBUG" "Checking the remains dns resource-records ..."
+echo "IBMCLOUD_DNS_INSTANCE_NAME: $IBMCLOUD_DNS_INSTANCE_NAME, BASE_DOMAIN: $BASE_DOMAIN"
+if [ ! -z ${IBMCLOUD_DNS_INSTANCE_NAME} ] && [ ! -z ${BASE_DOMAIN} ]; then
+  cmd="${IBMCLOUD_CLI} dns zones -i ${IBMCLOUD_DNS_INSTANCE_NAME} -o json | jq -r --arg n ${BASE_DOMAIN} '.[] | select(.name==\$n) | .id'"
+  dns_zone_id=$(eval "${cmd}")
+  cmd="${IBMCLOUD_CLI} dns resource-records ${dns_zone_id} -i ${IBMCLOUD_DNS_INSTANCE_NAME} | grep -w ${CLUSTER_FILTER}"
+  count=$(eval "${cmd} -c")
+  if [ "${count}" -gt 0 ]; then
+    echo "ERROR: remains dns resource-records..."
+    run_command "${cmd}"
+    hasError=1
+  fi
+fi
+
+echo "DEBUG" "Checking the remains cis dns-records ..."
+ibmcloud_cis_instance_name=$(cat "${CLUSTER_PROFILE_DIR}/ibmcloud-cis")
+if [ ! -z ${BASE_DOMAIN} ]; then
```

Comment:
Even it is a private cluster, also running this check? If I remember, the CIS domain is only used in a non-private cluster install.

Reply:
I plan to add the checking in all the workflow, 
the remains service instances, DNS records, and cis records

Reply:
My comments is not about where you are going to add the step, but private cluster does not need to check CIS, right?

Reply:
Of course, if you like to always run the check against a private cluster, I will let it go.